pkg/kubectl/resource/builder.go

```diff
@@ -584,6 +584,9 @@ func (b *Builder) visitorResult() *Result {
 			if b.flatten {
 				visitors = NewFlattenListVisitor(visitors, b.mapper)
 			}
+			if b.defaultNamespace {
+				visitors = NewDecoratedVisitor(visitors, SetNamespace(b.namespace))
+			}
```

Comment:
@caesarxuchao I think it's because other commands don't use FilenameParam() and Latest() together.